mono/metadata/mono-debug.h

```diff
@@ -227,6 +228,8 @@ MONO_API int             mono_debugger_insert_breakpoint           (const char *
 MONO_API void mono_set_is_debugger_attached (mono_bool attached);
 MONO_API mono_bool mono_is_debugger_attached (void);
 
+MONO_API void mono_add_assembly (MonoAssemblyLoadContext *alc, MonoAssembly *assembly, gpointer user_data, MonoError *error);
```

Comment:
Not reviewing the whole PR yet, but just a comment about this `MONO_API` function.

I would prefer not to make `add_assembly` a public API.
1. We don't have any other MonoAssemblyLoadContext public APIs yet.
2. The name is really general - it's not clear that this is for the debugger only.
3. It's signature has all of these arguments because it's a callback hook for the assembly loader.  If you need the same functionality for another purpose, I'd prefer if you refactored the hook - move the code to a new function with a more concise signature and call that from multiple places if you need to.
3. maybe this can be MONO_DEBUGGER_API (which doesn't exist) or something similar if a public symbol is really needed?

Reply:
You are right, it's not necessary to be a public symbol, it was a copy paste mistake.